package/MDAnalysis/lib/mdamath.py

```diff
@@ -110,7 +112,7 @@ def normal(vec1, vec2):
     return normal / n
 
 
-def pdot(a, b):
+def pdot(a: npt.NDArray, b: npt.NDArray) -> npt.NDArray:
```

Comment:
Automatic tools may tell you that, indeed, because it is how it is use in the examples it seen. But if you read the code, you see that `a` and `b`  are passed to the `np.einsum`function that expects an `ArrayLike` there.